internal/backend/remote-state/s3/client.go

```diff
@@ -308,10 +329,46 @@ func (c *RemoteClient) Lock(info *statemgr.LockInfo) (string, error) {
 			Err:  err,
 			Info: lockInfo,
 		}
-		return "", lockErr
+		return lockErr
 	}
 
-	return info.ID, nil
+	return nil
+}
+
+// s3Lock expects the statemgr.LockInfo#ID to be filled already
+func (c *RemoteClient) s3Lock(info *statemgr.LockInfo) error {
+	if !c.useLockfile {
+		return nil
+	}
+	info.Path = c.lockPath()
```

Comment:
Hmm, both codepaths update c.lockPath(), does it make sense to do that in the calling function?  It already knows if a lock should be acquired or not.

Reply:
Good catch. I extracted the `LockInfo` setup into the main entrypoint for locking (`Lock()`) but I forgot about this little line. Thanks for pointing it out.

Reply:
Fixed in 8c9faf7fca2641b352f9eae9127feee2b5901a20